olad/Olad.cpp

```diff
@@ -171,6 +175,11 @@ int main(int argc, char *argv[]) {
       ola::NewCallback(olad->GetOlaServer(), &ola::OlaServer::ReloadPlugins));
 #endif  // _WIN32
 
+#if HAVE_LIBSYSTEMD
+  // Return value is intentionally not checked. See return value section
+  // under sd_notify(3).
+  sd_notify(0, "READY=1\nSTATUS=Startup complete\n");
```

Comment:
Again, a bit of logging please.

Reply:
Hmm.. I don't think we need it for this one, though. It could be useful to tell the user that we're not running under `systemd`, or that we have failed to access the notification socket due to a configuration error.

However, if `olad` was launched as a systemd service of type `notify` (which is the intended use case): then `systemd` will report that the start of the service failed, anyway, which should alert the user to the error. There also shouldn't be an error accesing the notification socket, because `systemd` implicitly allows services with `type=notify` to access the notification socket.

Reply:
It's those edge cases that can be frustrating to troubleshoot though. If I forget to put type=notify, I want olad to tell me the socket isn't available, so I know to tweak my systemd config. Likewise in that timeout case, I'd like to know olad did/didn't send the message so I can troubleshoot ifdef type stuff.

Reply:
Yeah, I've added this in commit a5e57fe971e0a49f59bee5187923ef4bb5799894. A success/fail message everytime may clog up the logs if we want to use notify() a little more, though....